### PR TITLE
add zstandard as a run dependency

### PR DESCRIPTION
**Pull request by dholth**
### Description

Add zstandard as a runtime dependency (was transitively included via conda-package-handling / streaming). Used to fetch repodata.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

**Review by jezdez — APPROVED** [review]

File: docs/source/user-guide/concepts/conda-plugins.rst [hunk 1]
```diff
@@ -30,8 +30,7 @@ Below is an example of a very basic plugin "hook":
 
 
    @conda.plugins.hookimpl
-   def conda_subcommands():
-       ...
+   def conda_subcommands(): ...
```
[the remark this answers is not in the document]
Comment (kenodegard) [reply docs/source/user-guide/concepts/conda-plugins.rst:33]: no idea, just saw this in other PRs

[the remark this answers is not in the document]
Comment (kenodegard) [reply docs/source/user-guide/concepts/conda-plugins.rst:33]: https://docs.astral.sh/ruff/rules/pass-statement-stub-body/

**Review by kenodegard — APPROVED** [review]

File: docs/source/user-guide/concepts/conda-plugins.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (dholth) [reply docs/source/user-guide/concepts/conda-plugins.rst:33]: In the past only changed files fell under the linter rules.

[the remark this answers is not in the document]
Comment (kenodegard) [reply docs/source/user-guide/concepts/conda-plugins.rst:33]: that was back when we used `darker`, both `black` and `ruff` run the formatter on the entire code base

**Review by beeankha — APPROVED** [review]

**Review by jezdez — APPROVED** [review]

**Pull request merged**